tests/integration/s3/test_s3.py

```diff
@@ -2167,6 +2167,27 @@ def test_s3_get_object_header_overrides(self, s3_client, s3_bucket, snapshot):
         )
         snapshot.match("get-object", response)
 
+    def test_s3_header_value_boolean_to_lowercase(self, s3_client, s3_create_bucket):
+        bucket_name = "encrypted-bucket"
+        s3_create_bucket(Bucket=bucket_name)
+        s3_client.put_bucket_encryption(
+            Bucket=bucket_name,
+            ServerSideEncryptionConfiguration={
+                "Rules": [
+                    {
+                        "ApplyServerSideEncryptionByDefault": {
+                            "SSEAlgorithm": "AES256",
+                            "KMSMasterKeyID": "secret",
+                        }
+                    },
+                ]
+            },
+        )
+        response = s3_client.put_object(Bucket=bucket_name, Key="test-key", Body="test-value")
+        assert response["ResponseMetadata"]["HTTPHeaders"][
+            "x-amz-server-side-encryption-bucket-key-enabled"
+        ] in ["false", "true"]
```

Comment:
I think we should be stronger with the assertion here. Which value should it be in this case? true or false?